lib/active_merchant/billing/gateways/hsbc.rb

```diff
@@ -0,0 +1,210 @@
+require 'tmpdir'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class HsbcGateway < Gateway
+      self.test_url = 'https://devcluster.api.p2g.netd2.hsbc.com.hk/cmb-connect-payments-pa-collection-cert-proxy/v1/direct-debits/'
+      self.live_url = 'https://example.com/live' # TODO
+
+      self.supported_countries = ['HK']
+      self.default_currency = 'HKD'
+      self.supported_cardtypes = []
+
+      self.money_format = :cents
+
+      self.homepage_url = 'https://www.hsbc.com.hk/'
+      self.display_name = 'HSBC'
+
+      STANDARD_ERROR_CODE_MAPPING = {}
+
+      def initialize(options={})
+        requires!(options, :client_id, :client_secret, :profile_id, :public_key, :private_key)
+        super
+      end
+
+      def authorize(money, payment, options={})
```

Comment:
is `payment` needed here? Is it going to be used in a future? I think we don't use it in anyplace now.

Reply:
Nice catch! Fixed in https://github.com/waysact/active_merchant/pull/157/commits/09e9d5a9f704fe403e23459e41aa592781aa9046.